### PR TITLE
feat: pass extra dkg meta to fedimintd

### PR DESCRIPTION
**Pull request by dpc**
```
> env FM_EXTRA_DKG_META=foo=bar just mprocs
```

gives:

image

### CODE REVIEW

File: fedimintd/src/fedimintd.rs [hunk 1]
```diff
@@ -32,6 +33,8 @@ use crate::attach_default_module_init_params;
 /// Time we will wait before forcefully shutting down tasks
 const SHUTDOWN_TIMEOUT: Duration = Duration::from_secs(10);
 
+pub const FM_EXTRA_DKG_META_VAR: &str = "FM_EXTRA_DKG_META";
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:36]: I'm not sure about the name.

File: devimint/src/util.rs [hunk 2]
```diff
@@ -13,6 +14,24 @@ use tracing::{debug, warn};
...
+        let parts: Vec<&str> = pair.split('=').collect();
+        if parts.len() == 2 {
+            map.insert(parts[0].to_string(), parts[1].to_string());
+        } else {
+            return Err(format_err!("Invalid pair in map: {}", pair));
+        }
+    }
+    Ok(map)
+}
+
```
Comment (elsirion) [inline devimint/src/util.rs:34]: Can't you import this duplicated function from a common dependency of `fedimintd` and `devimint`?

Comment (dpc) [reply devimint/src/util.rs:34]: Me and me were debating it while implementing and reviewing it yesterday.

I don't know if we should bloat public interface of `fedimint-core` with every utility function that we happen to use in multiple places.

We probably ... want some `fedimint-internal` or `fedimint-utils-internal` that we could specifically keep low key, so external users don't use it. But even then we will have deal with it's semver, possibly by just incrementing major release like there's no tomorrow.

**Review by elsirion — APPROVED** [review]

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.1`:
- #3330

**Pull request merged**